store/tikv/oracle/oracle.go

```diff
@@ -22,14 +22,30 @@ import (
 	"go.uber.org/zap"
 )
 
+// Option represents available options for the oracle.Oracle.
+type Option struct {
+	TxnScope string
+}
+
+// NewDefaultOption will returns a default oracle.Oracle option.
+func NewDefaultOption() *Option {
+	return &Option{
+		// Todo: replace "global" with config.DefTxnScope
+		TxnScope: "global",
+	}
+}
+
+// OptionFunc configures the methods of oracle.Oracle.
+type OptionFunc func(*Option)
+
 // Oracle is the interface that provides strictly ascending timestamps.
 type Oracle interface {
-	GetTimestamp(ctx context.Context) (uint64, error)
-	GetTimestampAsync(ctx context.Context) Future
-	GetLowResolutionTimestamp(ctx context.Context) (uint64, error)
-	GetLowResolutionTimestampAsync(ctx context.Context) Future
-	IsExpired(lockTimestamp uint64, TTL uint64) bool
-	UntilExpired(lockTimeStamp uint64, TTL uint64) int64
+	GetTimestamp(ctx context.Context, opts ...OptionFunc) (uint64, error)
```

Comment:
Why not just pass `Option`?

Reply:
I think it's more reasonable to pass the `OptionFunc` to change the content of `Option` we create inside a method, which is more flexible and elegant.

Reply:
Currently, the code would be redundant for each caller. Why is it more elegant? Any assumed scenarios?